Orange/preprocess/impute.py

```diff
@@ -26,7 +27,61 @@ def transform(self, c):
         return numpy.where(numpy.isnan(c), self.value, c)
 
 
-class Average:
+class BaseImputeMethod:
+    name = ""
+    short_name = ""
+    description = ""
+    format = "{var.name} -> {short_name}"
+    columns_only = False
+
+    def __init__(self, *args, **kwargs):
+        pass
+
+    @abstract
+    def __call__(self, data, variable):
+        raise NotImplementedError()
```

Comment:
@abstract decorator will raise NotImplementedError. The body of the method could include a helpful docstring instead (describing what is the purpose of the method and its parameters).